tt_metal/impl/program/program.cpp

```diff
@@ -131,9 +130,22 @@ class Program_ {
     void allocate_circular_buffers(const Device *device);
 
     bool is_finalized() const;
+    void allocate_kernel_bin_buf_on_device(Device* device);
     void finalize(Device *device);
     bool is_cached() const { return this->cached_; }
+    ProgramBinaryStatus binaries_on_device_status(std::size_t device_id) const {
+        if (this->binaries_on_device_.find(device_id) != this->binaries_on_device_.end()) {
+            return this->binaries_on_device_.at(device_id);
+        }
+        return ProgramBinaryStatus::NotSent;
+    }
     void set_cached() { this->cached_ = true; }
+    void set_binaries_on_device_status(std::size_t device_id, ProgramBinaryStatus status) {
+        if (this->binaries_on_device_.find(device_id) != this->binaries_on_device_.end()) {
+            this->binaries_on_device_.at(device_id) = status;
+        }
```

Comment:
With `binaries_on_device_[device_id] = status;`, it will do the right thing if the element wasn't present in the map, so you can simplify the code.

Reply:
Thanks, updated